src/views/GovernanceReviewTeam/Actions/SubmitAction.tsx

```diff
@@ -12,30 +13,58 @@ import FieldGroup from 'components/shared/FieldGroup';
 import Label from 'components/shared/Label';
 import TextAreaField from 'components/shared/TextAreaField';
 import { ActionForm, ActionType, CreateActionPayload } from 'types/action';
+import { BasicActionInput } from 'types/graphql-global-types';
 import { postAction } from 'types/routines';
 import flattenErrors from 'utils/flattenErrors';
 import { actionSchema } from 'validations/actionSchema';
 
+type ActionInput = {
+  input: BasicActionInput;
+};
+
 type SubmitActionProps = {
   action: ActionType;
   actionName: string;
+  query: DocumentNode | null;
 };
 
-const SubmitAction = ({ action, actionName }: SubmitActionProps) => {
+const SubmitAction = ({ action, actionName, query }: SubmitActionProps) => {
   const { systemId } = useParams<{ systemId: string }>();
   const { t } = useTranslation('action');
   const history = useHistory();
   const dispatch = useDispatch();
 
+  let mutate: any;
+  let mutationResult: any;
+  if (query) {
```

Comment:
[Hooks shouldn't be called in conditionals](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level). 

It might be better to have `SubmitActionRedux` and `SubmitActionGraphQL` (there are definitely better names) and use one or the other at a higher level instead of having this branching within `SubmitAction`.

Reply:
@jim I like the idea of separating the components. Thanks for suggesting that!

Reply:
Alternatively, you could define the mutation outside the if statement, and then conditionally use it.